packages/core/test/oracle/GovernorV2.js

```diff
@@ -0,0 +1,923 @@
+    await voting.methods.revealVote(request.identifier, request.time, vote, salt).send({ from: accounts[0] });
+    await moveToNextRound(voting, accounts[0]);
+
+    // Verify execute event.
+    receipt = await governorV2.methods.executeProposal(id, 0).send({ from: accounts[0] });
+    await assertEventEmitted(receipt, governorV2, "ProposalExecuted", (ev) => {
+      return ev.id.toString() === id.toString() && ev.transactionIndex.toString() === "0";
+    });
+  });
+
+  it("No re-entrant execution", async function () {
+    // Send the proposal and verify that an event is produced.
+    const id = await governorV2.methods.numProposals().call();
+
+    // Construct the transaction that we want to re-enter and pass the txn data to the ReentrancyChecker.
+    const txnData = governorV2.methods.executeProposal(id.toString(), "0").encodeABI();
+    const reentrancyChecker = await ReentrancyChecker.new().send({ from: accounts[0] });
+    await reentrancyChecker.methods.setTransactionData(txnData).send({ from: accounts[0] });
+
+    // Propose the reentrant transaction.
+    await governorV2.methods
+      .propose([
+        {
+          to: reentrancyChecker.options.address,
+          value: 0,
+          data: constructTransferTransaction(account2, toWei("0")), // Data doesn't since it will hit the fallback regardless.
+        },
+      ])
+      .send({ from: accounts[0] });
+
+    // Vote the proposal through.
+    await moveToNextRound(voting, accounts[0]);
+    const roundId = await voting.methods.getCurrentRoundId().call();
+    const pendingRequests = await voting.methods.getPendingRequests().call();
+    const request = { ...pendingRequests[0], identifier: padRight(pendingRequests[0].identifier, 64) };
+    const vote = toWei("1");
+    const salt = getRandomSignedInt();
+    const hash = computeVoteHash({
+      price: vote,
+      salt,
+      account: proposer,
+      time: request.time,
+      roundId,
+      identifier: request.identifier,
+    });
+    await voting.methods.commitVote(request.identifier, request.time, hash).send({ from: accounts[0] });
+    await moveToNextPhase(voting, accounts[0]);
+    await voting.methods.snapshotCurrentRound(signature).send({ from: accounts[0] });
+    await voting.methods.revealVote(request.identifier, request.time, vote, salt).send({ from: accounts[0] });
+    await moveToNextRound(voting, accounts[0]);
+
+    // Since we're using the reentrancy checker, this transaction should FAIL if the reentrancy is successful.
+    await governorV2.methods.executeProposal(id, 0).send({ from: accounts[0] });
+  });
+
+  it("Starting id > 0", async function () {
+    // Set arbitrary starting id.
+    const startingId = 910284;
+
+    // Create new governorV2 contract.
+    finder = await Finder.deployed();
+    const newGovernor = await GovernorV2.new(finder.options.address, startingId, timer.options.address).send({
+      from: proposer,
+    });
+
+    const newVoting = await VotingV2.new(
```

Comment:
We also needed to create a new deployment of VotingV2 to be able to transfer the ownership to the new GovernorV2 deployment created in this test

Reply:
excellent. thanks for calling this out.